lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py

```diff
@@ -871,9 +869,8 @@ def compare(a, b, t):
             return True
         elif isinstance(a, dict) and isinstance(b, dict):
             for k in t.keys():
-                if not k == '__sort__':
-                    if not compare(a.get(k, None), b.get(k, None), t[k]):
-                        return False
+                if not compare(a.get(k, None), b.get(k, None), t[k]):
+                    return False
```

Comment:
line 875, what if a==none and b==none,  the compare return true, but it should be false.

Reply:
line 861, if s is not None, be specific s=="index"

Reply:
line 858-860, no matter what t is,  how many properties in it,  if one of properties =="index", s will be "index". sorry I didn't get it.

Reply:
line 887, default comparison, not  considering number